cross-edits-detection/cross-edit-suspicious-patterns.ts

```diff
@@ -0,0 +1,438 @@
+/**
+  Copyright 2019 Google LLC
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+**/
+
+import {CrossEditSuspiciousPatternsInfo, Revision} from "./interface";
+import {
+  DecisionLogProps,
+  DecisionLogDoc
+} from '~/shared/models/decision-log.model';
+
+export class CrossEditSuspiciousPatterns implements Revision {
```

Comment:
It seems a lot of duplication with CrossEditSuspiciousPatternInfos. why do we need `CrossEditSuspiciousPatternsInfos` if we already have `CrossEditSuspiciousPatterns`? Also, this name is not descriptive, can you change it to something else?